test/edu/stanford/dlss/wowza/TestSulWowza.java

```diff
@@ -0,0 +1,630 @@
+
+    @Test
+    public void getFilename()
+    {
+        assertEquals("stream.mp4", testModule.getFilename("oo/00/oo/0000/stream.mp4"));
+        assertEquals("anything.without%slash", testModule.getFilename("a/b/c/d/anything.without%slash"));
+        assertNull(testModule.getFilename("a/b/c/d/"));
+    }
+
+    @Test
+    public void getVerifyStacksTokenUrl_validUrl()
+    {
+        WMSProperties mockProperties = mock(WMSProperties.class);
+        when(mockProperties.getPropertyStr(anyString(), anyString())).thenReturn(SulWowza.DEFAULT_STACKS_TOKEN_VERIFICATION_BASEURL);
+        IApplicationInstance appInstanceMock = mock(IApplicationInstance.class);
+        when(appInstanceMock.getProperties()).thenReturn(mockProperties);
+        testModule.onAppStart(appInstanceMock);
+        String druid = "oo000oo0000";
+        String filename = "filename.ext";
+        String userIp = "0.0.0.0";
+        String expPath = "/media/" + druid + "/" + filename + "/verify_token";
+        String expQueryStr = "?stacks_token=" + stacksToken + "&user_ip=" + userIp;
+        String expUrl = SulWowza.DEFAULT_STACKS_TOKEN_VERIFICATION_BASEURL + expPath + expQueryStr;
+
+        String resultUrl = testModule.getVerifyStacksTokenUrl(stacksToken, druid, filename, userIp);
+        assertEquals(expUrl, resultUrl);
+    }
+
+    @Test
+    /** expect it to return null and log an error message */
+    public void getVerifyStacksTokenUrl_malformedUrl()
+    {
+        WMSProperties mockProperties = mock(WMSProperties.class);
+        when(mockProperties.getPropertyStr(anyString(), anyString())).thenReturn("badUrl");
+        IApplicationInstance appInstanceMock = mock(IApplicationInstance.class);
+        when(appInstanceMock.getProperties()).thenReturn(mockProperties);
+        testModule.onAppStart(appInstanceMock);
+
+        // logger is the rootLogger, per test/resources/log4j.properties
+        Logger logger = Logger.getRootLogger();
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        Layout layout = new SimpleLayout();
+        Appender appender = new WriterAppender(layout, out);
+        logger.addAppender(appender);
+
+        try
+        {
+            String resultUrl = testModule.getVerifyStacksTokenUrl(stacksToken, "oo000oo0000", "filename.ext", "0.0.0.0");
+            assertNull(resultUrl);
+            String logMsg = out.toString();
+            assertThat(logMsg, allOf(containsString("ERROR"),
+                                     containsString(" bad URL for stacks_token verification:"),
+                                     containsString(" java.net.MalformedURLException"),
+                                     containsString(" no protocol")));
+        }
+        finally
+        {
+            logger.removeAppender(appender);
+        }
+    }
+
+    //@Test
+    public void verifyStacksToken_wUrlString_logsRequestMade()
+    {
+        fail("don't know how to test this without spinning up a local stacks instance");
```

Comment:
Sorry to be piling on even more things, but could we just mock the connection? Something like:

HttpURLConnection spyStacksConnection = spy(new HttpURLConnection)
URL mockURL = mock(URL.class)
when(mockURL.new("whatever URL we are testing").thenReturn(spyStacksConnection)
...
verify that the various http connection methods are called
...

I'm happy to pair on this later if you like.